pkg/elasticsearch/indices.go

```diff
@@ -31,6 +31,79 @@ func (ec *esClient) CreateIndex(name string, index *estypes.Index) error {
 	return nil
 }
 
+func (ec *esClient) UpdateIndexSettings(name string, settings *estypes.IndexSettings) error {
+	body, err := utils.ToJson(settings)
+	if err != nil {
+		return err
+	}
+	payload := &EsRequest{
+		Method:      http.MethodPut,
+		URI:         fmt.Sprintf("%s/_settings", name),
+		RequestBody: body,
+	}
+	logger.DebugObject("UpdateIndexSettings with payload: %#v", settings)
+	ec.fnSendEsRequest(ec.cluster, ec.namespace, payload, ec.k8sClient)
+	if payload.Error != nil {
+		return payload.Error
```

Comment:
Consider refactor response evaluation into common function

Reply:
Let's keep it out for now. #356 is already big enough for landing in release 4.5